x-pack/plugins/cases/server/plugin.ts

```diff
@@ -104,6 +105,8 @@ export class CasePlugin {
       )}] and plugins [${Object.keys(plugins)}]`
     );
 
+    registerInternalAttachments(this.externalReferenceAttachmentTypeRegistry);
```

Comment:
If this is an "Internal attachment type" that should always exist maybe it would make more sense to register it when `externalReferenceAttachmentTypeRegistry` is initialized and not when the cases plugin is set up?

Reply:
I think I'm open to either keeping the registration in the plugin or making it a part of the `ExternalReferenceAttachmentTypeRegistry`. If it makes testing easier to do registration in the plugin that's fine with me.

Reply:
> It will also make the testing much harder if the state of the class contains our internal attachments.

But if I understand it correctly the class is "ours" so wherever it is used it will need those internal attachments.

My reasoning was that it is why more error-prone to have to remember to register internal types every time we instantiate it. External ones though are fine because those are the responsibility of the "instantiator".

Reply:
> My reasoning was that it is why more error-prone to have to remember to register internal types every time we instantiate it

I see what you mean. This should never happen. This class is instantiated only once in the `setup` lifecycle method of the cases plugin. The same instance is passed down to the cases code and lives throughout the whole lifecycle of the cases plugin (This can be a counterargument to my main point 🙂). If you need a new instance of the class you probably need it for testing or to do something else unrelated to the internal attachments.

I see the cases plugin as an external consumer of the `externalReferenceAttachmentTypeRegistry` and not as an internal consumer. By registering data in the constructor of the `externalReferenceAttachmentTypeRegistry` you tight couple the date with the class. The class is now aware of the data which are implementation detail. The class has more concerns than it should have.